specification/healthdataaiservices/HealthDataAIServices.DeidServices/models_document_flow.tsp

```diff
@@ -0,0 +1,194 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "./models.tsp";
+
+using Azure.ClientGenerator.Core;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using Azure.Core;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "AI Casing style to match OpenAI"
+namespace HealthDataAIServices.DeidServices;
+
+@doc("A job containing a batch of documents to de-identify.")
+@resource("jobs")
+model DeidentificationJob {
+  @key
+  @doc("The name of a job.")
+  @visibility("read")
+  @maxLength(36)
+  @minLength(3) // Must be 3 to match regex
+  @pattern("^[a-zA-Z0-9][a-zA-Z0-9-_]+[a-zA-Z0-9]$")
+  name: string;
+
+  @doc("Operation to perform on the input documents.")
+  operation?: DeidentificationOperationType = DeidentificationOperationType.Surrogate;
+
+  @doc("Storage location to perform the operation on.")
+  sourceLocation: SourceStorageLocation;
+
+  @doc("Target location to store output of operation.")
+  targetLocation: TargetStorageLocation;
+
+  @doc("Customization parameters to override default service behaviors.")
+  customizations?: DeidentificationJobCustomizationOptions;
+
+  // Readonly Properties
+
+  @doc("Current status of a job.")
+  @visibility("read")
+  @pollingOperationParameter
+  status: DeidentificationJobStatus;
+
+  @doc("Error when job fails in it's entirety.")
+  @visibility("read")
+  error?: Azure.Core.Foundations.Error;
+
+  @doc("""
+    Date and time when the job was completed.
+    
+    If the job is canceled, this is the time when the job was canceled.
+    
+    If the job failed, this is the time when the job failed.
+    """)
+  @visibility("read")
+  lastUpdatedAt: utcDateTime;
+
+  @doc("Date and time when the job was created.")
+  @visibility("read")
+  createdAt: utcDateTime;
+
+  @doc("Date and time when the job was started.")
+  @visibility("read")
+  startedAt?: utcDateTime;
+
+  @doc("Summary of a job. Exists only when the job is completed.")
+  @visibility("read")
+  summary?: DeidentificationJobSummary;
+}
+
+@doc("List of statuses a job can have.")
+@lroStatus
+union DeidentificationJobStatus {
+  // Cannot use Azure.Core.Foundations.OperationState. because it is not a union type.
+  @doc("Job has been submitted and is waiting to be processed.")
+  NotStarted: "NotStarted",
+
+  @doc("Job has been started.")
+  Running: "Running",
+
+  @lroSucceeded
+  @doc("Job has completed successfully. All documents have succeeded.")
+  Succeeded: "Succeeded",
+
+  @lroFailed
+  @doc("Job has completed with at least a single document failing.")
+  PartialFailed: "PartialFailed",
```

Comment:
Yes it is a new terminal state. 

Hmmmm, I haven't experienced any problems in the SDK while using LROs.

If we have a job that contains 1000 files and we have 500 successes and 500 failures, do you suggest we just return `Failed` instead of `PartialFailed`? Currently we return failed if all files fail, or there is some underlying prerequisite check that doesn't succeed. Partial Failed is for we had some files failed and some succeed.

Reply:
@mikekistler 

This appears to be API similar to a batch processing. Any advice on the terminal state for a partial success?

Reply:
Agreed that we should not add a new terminal state. I think "Failed" is probably the best option. For "failed", users would need to examine each of the result values to determine which failed. With "Succeeded", all succeeded so this extra processing is bypassed.